src/shell/doom.c

```diff
@@ -126,9 +126,9 @@ static void doom_check (struct shell_doom *doom,
 
     if (doom->exit_on_error && doom->exit_rc != 0) {
         shell_die (doom->exit_rc,
-                   "%s: %srank %d on host %s failed and exit-on-error is set",
+                   "%s: %s rank %d on host %s failed and exit-on-error is set",
                    get_jobspec_command_arg0 (doom),
-                   doom->lost_shell ? "shell " : "",
+                   doom->lost_shell ? "shell" : "task",
```

Comment:
nit: "task" -> "task on"?  there's only one shell but there can be multiple tasks?